File: solutions/haskell/armstrong-numbers/1/src/ArmstrongNumbers.hs
module ArmstrongNumbers (armstrong) where

to_digits :: Integral x => x -> [x]
to_digits 0 = []
to_digits x = x `mod` 10 : to_digits (x `div` 10)

armstrong :: Integral a => a -> Bool
armstrong a = do
  let digits = to_digits a
  let number_of_digits = length digits
  a == sum [d ^ number_of_digits | d <- digits]
  

File: solutions/python/armstrong-numbers/1/armstrong_numbers.py
def is_armstrong_number(number: int) -> bool:
    """
    Check if a number is an Armstrong number.
    An Armstrong number is equal to the sum of its digits
    raised to the power of the number of digits.
    """
    s = str(number)
    power = len(s)
    return number == sum(int(ch) ** power for ch in s)